src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -386,6 +389,24 @@ def nonhermetic_digest_classpath():
       # Update the products with the latest classes.
       self.register_extra_products_from_contexts([ctx.target], compile_contexts)
 
+    def work_for_vts_cachewrite(vts, ctx):
+      # Double check the cache before beginning compilation
+      hit_cache = self.check_cache(vts, counter)
+      target = ctx.target
+
+      if not hit_cache:
+        counter_val = str(counter()).rjust(counter.format_length(), ' ')
+        counter_str = '[{}/{}] '.format(counter_val, counter.size)
+        self.context.log.info(
+          counter_str,
+          'Writing to cache for ',
+          items_to_report_element(ctx.sources, '{} source'.format(self.name())),
+          ' in ',
+          items_to_report_element([t.address.reference() for t in vts.targets], 'target'),
+          ' (',
+          ctx.target.address.spec,
+          ').')
```

Comment:
Python 3 allows [f-strings](https://stackoverflow.com/questions/43123408/f-strings-in-python-3-6), which make this type of formatting much easier to follow. Additionally, I believe the elements such as `items_to_report_element(ctx.sources, '{} source'.format(self.name()))` are repeated elsewhere in this file as well. Is it possible to factor out that message templating into its own function?